src/boosting/gbdt.cpp

```diff
@@ -547,15 +547,18 @@ void GBDT::SaveModelToFile(int num_iteration, const char* filename) const {
     output_file << pairs[i].second << "=" << std::to_string(pairs[i].first) << std::endl;
   }
 
-  output_file << std::endl << "feature information:" << std::endl;
-  for (size_t i = 0; i < max_feature_idx_ + 1; ++i) {
-    output_file << feature_names_[i] << "=" << feature_infos_[i] << std::endl;
+  if (feature_infos_.size() == feature_names_.size()) {
+      output_file << std::endl << "feature information:" << std::endl;
+      for (size_t i = 0; i < max_feature_idx_ + 1; ++i) {
+        output_file << feature_names_[i] << "=" << feature_infos_[i] << std::endl;
+      }
```

Comment:
@guolinke About this issue, I can do a PR, but it looks as if it doesn't make sense to load `feature_infos_`, as the training data may not be available in the GBDT class when the model is loaded.

Therefore, I would rather skip saving `feature_infos_` if it's empty, as it would mean that the model was loaded and there is no training data available to which it would be useful.